### PR TITLE
Refactor: Remove code from package main

### PR DESCRIPTION
**Pull request by mcpherrinm**
Code in `package main` is harder to test, cannot be imported easily, and
hurts modularity.

The codestyle here is:
package main is in
 github.com/square/sharkey/cmd/sharkey-server
 github.com/square/sharkey/cmd/sharkey-agent

Those can be "go get"ted, and will build binaries the same as always.

All other code is migrated into the pkg/ directory, both for server and client.
There's a "cli" package under pkg/client and pkg/server that handles parsing
command line options and loading config files.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 language: go
 
-sudo: required
-dist: trusty
```
Comment (mcpherrinm) [inline .travis.yml]: trusty is old; switch to default/current bionic.  `sudo:` is obsolete, and is always provided now.

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-
 services:
+  - mysql
```
Comment (mcpherrinm) [inline .travis.yml]: mysql isn't provided by default post-trusty.

File: .travis.yml (hunk 1 above)
Comment (mbyczkowski) [reply .travis.yml]: 👍

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,5 @@
-sharkey-server
-sharkey-client
-/server/server
-/client/client
+/sharkey-server
```
Comment (mbyczkowski) [inline .gitignore]: oops, I think it was me 😅

File: .travis.yml (hunk 2 above)
Comment (mbyczkowski) [reply .travis.yml]: 👍

File: .travis.yml [hunk 4]
```diff
@@ -14,18 +12,22 @@ cache:
...
-install:
-  - go build -o sharkey-server -v ./server
-  - go build -o sharkey-client -v ./client
-  - docker build -t server:latest -f Dockerfile .
-  - docker build -t client:latest -f DockerfileClientTest .
+
+before_install:
+  - mysql -e 'CREATE DATABASE sharkey_test;'
+
+install: skip
```
Comment (mbyczkowski) [inline .travis.yml]: just curious why are we skipping this? I guess there's nothing to do if everything is in `script`?

File: .travis.yml [hunk 5]
```diff
@@ -14,18 +12,22 @@ cache:
...
 
 script:
+  # Docker images
```
Comment (mbyczkowski) [inline .travis.yml]: at some point we could just put everything in `ci.sh` or something like that

File: cmd/sharkey-server/sharkey-server.go [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package main
```
Comment (mbyczkowski) [inline cmd/sharkey-server/sharkey-server.go]: this looks really neat. Nice! 💯

File: .travis.yml (hunk 4 above)
Comment (mcpherrinm) [reply .travis.yml]: There's a default "install" step which installs your dependencies (via godep or other stuff).  We don't need that as we're using `go mod` to fetch on the fly.  It's probably harmless to just do nothing, but I wanted to make sure it wasn't messing with my build.

File: .travis.yml (hunk 5 above)
Comment (mcpherrinm) [reply .travis.yml]: Yeah; and maybe doing docker + integration tests in a different build from go tests & binaries.

**Pull request merged**